examples/basic-usage.py

```diff
@@ -0,0 +1,237 @@
+# 'neighbor shifts' as is indicated by the quantity 'S'. This is because we are
+# typically still interested in taking gradients with respect to the atomic positions.
+# An exact definition can be found in the `vesin` documentation.
+
+nl = NeighborList(cutoff=rcut, full_list=False)
+i, j, neighbor_distances = nl.compute(
+    points=positions, box=cell, periodic=True, quantities="ijd"
+)
+neighbor_indices = torch.stack([i, j], dim=1)
+
+# %%
+#
+# Main Part: Calculator
+# ---------------------
+#
+# The `Calculator` classes are the main user-facing classes in `torchpme`. These are
+# used to compute atomic potentials $V_i$ for a given set of positions and particle
+# weights (charges). For periodic calculators that are the main focus of this tutorial,
+# it is also required to specify a `cell`.
+#
+# We start by the initialization of the class.
+
+calculator = EwaldCalculator(potential=potential, lr_wavelength=lr_wavelength)
+
+# %%
+#
+# Compute Energy
+# --------------
+#
+# We have now all ingredients: we can use the `Calculator` class to, well, actually compute the potentials $V_i$ at the position of the atoms, or the total energy for the given particle weights (charges). The electrostatic potential can then be obtained as
+# .. math::
+#
+#   E = \sum_{i=1}^N q_i V_i
+
+potentials = calculator.forward(
+    charges, cell, positions, neighbor_indices, neighbor_distances
+)
+energy = torch.sum(charges * potentials)
+
+print(energy)
+
+# %%
+#
+# Compute Forces using backpropagation (automatic differentiation)
+# ----------------------------------------------------------------
+#
+# The forces on the particles can simply be obtained as minus the gradient of the energy
+# with respect to the positions. These are easy to evaluate using the automatic
+# differentiation capabilities of `pytorch` using the backpropagation method.
+#
+# note that this only works since we set `requires_grad=True` when we initialized the positions tensor above.
+
+energy.backward()
+forces = -positions.grad
+
+print(forces[:3])
+
+# %%
+#
+# Aperiodic Structures
+# --------------------
+#
+# For now, we have been using the `EwaldCalculator` which is a periodic calculator. We
+# can however also use it for aperiodic structures by just using it as a calculator with
+# a cutoff radius.
+
+# Clone the positions to avoid accumulating gradients with respect to the same
+# variables multiple times
+positions_aperiodic = positions.clone().detach()
+positions_aperiodic.requires_grad = True
+
+# Compute neighbor list but this time without periodic boudary conditions
+i, j, neighbor_distances_aperiodic = nl.compute(
+    points=positions_aperiodic, box=cell, periodic=True, quantities="ijd"
+)
+neighbor_indices_aperiodic = torch.stack([i, j], dim=1)
+
+# Compute aperiodic potential
+potentials_aperiodic = calculator._compute_rspace(
+    charges, neighbor_indices_aperiodic, neighbor_distances_aperiodic
+)
```

Comment:
I agree, but also feel that this should be a separate PR. I can open an issue about this. Until then, we can either keep this example using the private function, or even just remove it altogether.

Reply:
I think you can already do it 

```python
from torchpme import Calculator

calculator = Calculator(potential=CoulombPotential(smearing=0))
calculator.forward(...)
```

even though I agree the same is not perfect;y indicating that this is for aperiodic structures.

Reply:
This doesn't work, because in its current version, `Calculator`'s `forward` requires an implementation of the reciprocal space sum. So this just raises an error. I have kept the example for now, but made the comments more detailed to reduce potential confusion about why this has been implemented like this.